datafusion/core/src/datasource/listing_table_factory.rs

```diff
@@ -63,16 +63,29 @@ impl TableProviderFactory for ListingTableFactory {
             ))?
             .create(session_state, &cmd.options)?;
 
-        let file_extension = get_extension(cmd.location.as_str());
+        let mut table_path = ListingTableUrl::parse(&cmd.location)?;
+        let file_extension = match table_path.is_collection() {
+            true => "",
+            false => &get_extension(cmd.location.as_str()),
+        };
```

Comment:
This behavior actually feels a bit odd to me, but was necessary to preserve the existing unit tests related to building the appropriate glob pattern for compressed (e.g. '.csv.gz') files. It seems more correct to use the default file extension in the case one doesn't need to be set explicitly, however that would be a small refactor and I didn't want to mix strict bug fixes with subtle refactors without additional input from reviewers.

Reply:
Can you please also add a comment explaining the rationale for the change?

I agree it looks odd -- I haven't studied the code but it feels like maybe this is treating the symptom of some larger problem and a real fix elsewhere would be simpler